graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/federation/FederatedSchemaGeneratorHooks.kt

```diff
@@ -82,12 +83,14 @@ open class FederatedSchemaGeneratorHooks(private val federatedTypeRegistry: Fede
             // SDL returned by _service query should NOT contain
             // - default schema definition
             // - empty Query type
-            // - directives
-            // - new scalars
+            // - any directive definitions
+            // - any custom directives
+            // - new federated scalars
             val sdl = originalSchema.print(includeDefaultSchemaDefinition = false)
-                .replace(directiveRegex, "")
-                .replace(scalarRegex, "")
-                .replace(emptyQuery, "")
+                .replace(directiveDefinitionRegex, "")
```

Comment:
Can we add a comment here that this can be simplified once this is solved: https://github.com/apollographql/apollo-server/issues/3334

Reply:
oke @smyrick wait

Reply:
Done @smyrick :)